### ISSUE TITLE
i18n: Translation system ideas

### ISSUE THREAD

**Issue by xaur** [decred/dcrdex#1880]
I got some ideas for the translation system while scanning DCRDEX Git history for the [status dashboard](https://decredcommunity.github.io/translations/status#dcrdex). Just wanted to share it somewhere, feel free to close at any point if nothing can be reasonably implemented.

1\. Document which branch to translate.

Most translations were made in `master` and then ported to `release-*`, but recently one was first [submitted](https://github.com/decred/dcrdex/pull/1815) to a release branch and then [ported](https://github.com/decred/dcrdex/pull/1831) to `master`. I see a room for confusion here. It may be worth updating the [Localization and Translation](https://github.com/decred/dcrdex/blob/master/docs/wiki/Localization-and-Translation.md) to instruct translators which branch to checkout and translate (and rely on maintainers porting to the other one), or do both.

2\. Put each language in its own file(s).

It is hard to follow Git history for a specific language translation when multiple languages live in the same file in [locale_ntfn.go](https://github.com/decred/dcrdex/blob/94a9aef6442c77cd7a55d5b136a1a56f125f09b8/client/core/locale_ntfn.go) and [locales.ts](https://github.com/decred/dcrdex/blob/94a9aef6442c77cd7a55d5b136a1a56f125f09b8/client/webserver/site/src/js/locales.ts). `git log` for these paths shows each change in each language (including English) and also in the extra code sitting there. The [locales/*.go](https://github.com/decred/dcrdex/tree/master/client/webserver/locales) files are much easier to track because each file has only one language and no extra processing code.

Originally I missed this to determine how up-to-date the translations are, but placing English strings separately from other languages *and* code also simplifies tracking changes in *English* strings using normal Git/GitHub tools. Translators could potentially see what exactly they need to sync (sometimes only tiny diffs).

3\. Extract translations into JSON files.

Originally adopted [in Decrediton](https://github.com/decred/decrediton/tree/master/app/i18n/translations), storing translations in simple key-value .json files allowed to build a GUI assist tool:

![decrediton-translator-zh](https://user-images.githubusercontent.com/24484727/193345203-4a19d796-916b-4b9e-9382-e28cbdff5c40.png)

It guides you through English strings that need to be (re-)translated and shows diffs of what's changed (most helpful for spotting small changes in large strings like in the screenshot above). The same method was recently [adopted](https://github.com/decred/dcrweb/tree/master/transifex_catalogs) by dcrweb, proving it can serve more than one project.

In DCRDEX strings are currently stored in .go and .ts files. This is easier to consume from the code side (esp. where the strings are wired to code's constants or enums), but harder to fit into the tool above.

It was mentioned [in Matrix](https://matrix.to/#/!qYctNXtSHVuWpHwZrK:decred.org/$1f0MPXQykZ2PEoeOwjtCQKV8Xvj-F_oDUfpWH57J_18) that there are challenges to arranging strings this way in DCRDEX, but maybe someone will have an idea how to overcome them. For example, write a small tool to generate .go files with proper const->string mappings from .json files. Or generalize the new [checker tool](https://github.com/decred/dcrdex/pull/1835) to compare arbitrary .json maps.

**Comment by vctt94** [comment]
I can work at suggestions 2/3. For notifications, at [locale_ntfn.go](https://github.com/decred/dcrdex/blob/94a9aef6442c77cd7a55d5b136a1a56f125f09b8/client/core/locale_ntfn.go) I guess it will be more trick, but I can see what can be done

**Comment by buck54321** [comment]
@vctt94 We need to discuss this on Element before moving forward.
